p2p/src/main/kotlin/maru/p2p/MaruPeerManager.kt

```diff
@@ -152,26 +154,32 @@ class MaruPeerManager(
         if (p2pNetwork.peerCount >= maxPeers || connectionInProgress.contains(peer.nodeIdBytes)) {
           return
         }
-        connectionInProgress.add(peer.nodeIdBytes!!)
+        connectionInProgress.add(peer.nodeIdBytes)
       }
 
-      log.debug("Peer {} Connecting to peer {}...", discoveryService!!.getLocalNodeRecord().nodeId, peer.nodeIdBytes)
+      log.debug("peer={} Connecting to peer={}...", discoveryService!!.getLocalNodeRecord().nodeId, peer.nodeIdBytes)
       p2pNetwork
         .connect(p2pNetwork.createPeerAddress(peer))
         .orTimeout(30, TimeUnit.SECONDS)
         .whenComplete { _, throwable ->
           try {
             if (throwable != null) {
-              log.debug("Failed to connect to peer {}: {}", peer.nodeIdBytes, throwable.stackTraceToString())
+              log.error("Failed to connect to peer ${peer.nodeIdBytes}", throwable)
```

Comment:
please favour placeholders. 

```suggestion
              log.error("Failed to connect to peer={}", peer.nodeIdBytes,  throwable)
```